crates/volta-core/src/project.rs

```diff
@@ -133,43 +114,19 @@ impl Project {
 
     /// Returns true if the input binary name is a direct dependency of the input project
     pub fn has_direct_bin(&self, bin_name: &OsStr) -> Fallible<bool> {
-        let dep_bins = self.dependent_bins.get(&self)?;
-        if let Some(bin_name_str) = bin_name.to_str() {
-            if dep_bins.contains_key(bin_name_str) {
-                return Ok(true);
+        if let Some(name) = bin_name.to_str() {
+            let config_path = path::user_tool_bin_config(name)?;
+            if config_path.exists() {
+                let config = BinConfig::from_file(config_path)?;
+                return Ok(self.has_direct_dependency(&config.package));
             }
         }
         Ok(false)
     }
 
-    /// Returns a mapping of the names to paths for all the binaries installed
-    /// by direct dependencies of the current project.
-    fn dependent_binaries(&self) -> Fallible<HashMap<String, String>> {
-        let mut dependent_bins = HashMap::new();
-        let all_deps = Manifest::for_dir(&self.project_root)?.merged_dependencies();
-        let all_dep_paths = all_deps.iter().map(|name| self.get_dependency_path(name));
-
-        // use those project paths to get the "bin" info for each project
-        for pkg_path in all_dep_paths {
-            let pkg_info =
-                Manifest::for_dir(&pkg_path).with_context(|_| ErrorDetails::DepPackageReadError)?;
-            let bin_map = pkg_info.bin;
-            for (name, path) in bin_map.iter() {
-                dependent_bins.insert(name.clone(), path.clone());
-            }
-        }
-        Ok(dependent_bins)
-    }
-
-    /// Convert dependency names to the path to each project.
-    fn get_dependency_path(&self, name: &String) -> PathBuf {
-        // ISSUE(158): Add support for Yarn Plug'n'Play.
-        let mut path = PathBuf::from(&self.project_root);
-
-        path.push("node_modules");
-        path.push(name);
-
-        path
+    pub fn has_direct_dependency(&self, dependency: &str) -> bool {
```

Comment:
It *looks* like this is only `pub` so it can be tested. If so, you can actually drop the `pub` marker entirely: the submodule has access to its parent module's private items. You can see an example of how that works [in this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4d494347badb5a86bcbced056f00744a).

Reply:
Good catch, updated!